### ISSUE TITLE
Question on completion category choice

### ISSUE THREAD

**Issue by bdarcus** [oantolin/embark#179]
Currently the code I'm working on uses simply "bibtex" as the completion category.

But in [this comment](https://github.com/oantolin/embark/issues/3#issuecomment-629728652), @oantolin notes emacs has two internal bibtex-related categories:

- bibtex-key
- bibtex-string

Does it even matter for this purpose? What are the implications of the three choices?

That using a standard category may end up associating additional commands with these candidates?

Bottom line, should one generally prefer the standard category?

**Comment by oantolin** [comment]
I have no idea what Emacs uses those categories for. I remember I found them simply by grepping the source for `category`. So the first step would be to find them again in the source and see what they are used for.

**Comment by oantolin** [comment]
Found 'em, they are bth used in `bibtex-completion-at-point-function` in `/usr/share/emacs/#/lisp/textmodes/bibtex.el` (where `#` stands for your Emacs version number). I'm still not sure what each one is for.

**Comment by bdarcus** [comment]
Hmm ... am I right, or is that not very well-documented? I can't figure it out ATM either, but will take another look later.

I can say that mode is for editing and validating bibtex files, while bibtex-completion is of course broader. It includes a function to open a bibtex entry given a key, for example.

Sounds like maybe I should just keep things as they are.

I guess the broader question is if any suggestions on how to define categories?

**Comment by oantolin** [comment]
> Sounds like maybe I should just keep things as they are.

I agree.

> I guess the broader question is if any suggestions on how to define categories?

I guess in the context of Marginalia and Embark, the main thing you should ask yourself when deciding whether to use an existing category name or a new one is whether you want the same annotations and actions available in the places where the exiting name is used. In this case you probably don't, since, for example, inserting the full bibtex entry in the bibtex file inside an entry makes no sense.

**Comment by bdarcus** [comment]
Got it. Thank you.
